core/src/main/kotlin/fr/sncf/osrd/standalone_sim/ScheduleMetadataExtractor.kt

```diff
@@ -546,6 +516,149 @@ fun routingRequirements(
     return res
 }
 
+private fun getRouteCriticalPos(
+    fullInfra: FullInfra,
+    routePath: StaticIdxList<Route>,
+    blockPath: StaticIdxList<Block>,
+    blockOffsets: OffsetArray<TravelledPath>,
+    zoneCount: Int,
+    signalingTrainStates: Map<LogicalSignalId, SignalingTrainState>,
+    envelope: EnvelopeInterpolate,
+    etcsSimulator: ETCSBrakingSimulator?,
+    routeStartBlockIndex: Int,
+    firstRouteBlock: BlockId,
+): Offset<TravelledPath>? {
+    val blockInfra = fullInfra.blockInfra
+    val simulator = fullInfra.signalingSimulator
+
+    val sigSystemId = blockInfra.getBlockSignalingSystem(firstRouteBlock)
+    val isCurveBased = simulator.sigModuleManager.isCurveBased(sigSystemId)
+    return if (isCurveBased) {
+        if (
+            simulator.sigModuleManager.getName(sigSystemId) != ETCS_LEVEL2.id ||
+                etcsSimulator == null
+        ) {
+            TODO(
+                "Routing requirements for curve-based signal are only available for " +
+                    "ETCS_LEVEL2 and through StandaloneSimulation"
+            )
+        }
+        getEtcsRouteCriticalPos(
+            blockInfra,
+            blockOffsets,
+            envelope,
+            etcsSimulator,
+            routeStartBlockIndex,
+            firstRouteBlock,
+        )
+    } else {
+        getSightRouteCriticalPos(
+            fullInfra,
+            routePath,
+            blockPath,
+            blockOffsets,
+            signalingTrainStates,
+            zoneCount,
+            routeStartBlockIndex,
+        )
+    }
+}
+
+private fun getEtcsRouteCriticalPos(
+    blockInfra: BlockInfra,
+    blockOffsets: OffsetArray<TravelledPath>,
+    envelope: EnvelopeInterpolate,
+    etcsSimulator: ETCSBrakingSimulator,
+    routeStartBlockIndex: Int,
+    firstRouteBlock: BlockId,
+): Offset<TravelledPath> {
+
+    // Get the braking curve for the first block of the route
+    val signalOffset =
+        blockOffsets[routeStartBlockIndex] +
+            blockInfra.getSignalsPositions(firstRouteBlock).first().distance
+
+    val singleEnvelope = envelope.rawEnvelopeIfSingle
+    // TODO: stop using a single envelope that's unavailable in STDCM and maybe move to a
+    //   dedicated EnvelopeTimeInterpolate.getIntersection().
+    //   Then probably protect its failure by returning
+    //   SignalRequirementsCreationStatus.NOT_SEEN_IN_AVAILABLE_SIM.
+    assert(singleEnvelope != null) {
```

Comment:
We're in `ScheduleMetadataExtractor`, it's not compatible with incremental paths anyway. We can pass the envelope directly as input to `getRouteCriticalPos`.

But (unrelated): we'll already have to deeply modify `IncrementalPath` soon for backtracks. Maybe we could look into turning it into a kind of "incremental simulation", including simulation elements? I don't know, might be quite tricky.

Reply:
Oh I now realize I misunderstood some things. It's not about having an envelope covering the whole path, but one specifically of type `Envelope` and not the interface.

Anyway. It's good if you find this new version better than before, but I'm resolving this either way 😅